internal/core/key_test.go

```diff
@@ -34,86 +34,6 @@ func TestNewDataStoreKey_ReturnsEmptyStruct_GivenEmptyString(t *testing.T) {
 	assert.ErrorIs(t, ErrEmptyKey, err)
 }
 
-func TestNewDataStoreKey_ReturnsCollectionIdAndIndexIdAndDocIDAndFieldIdAndInstanceType_GivenFourItemsWithType(
```

Comment:
question: Do we loose any coverage or paths that these tests were testing due to the removal?

Reply:
I doubt it, in theory everything should be covered by integration tests anyway, and the new code has fewer branches IIRC.